android/src/main/res/layout/card_details.xml

```diff
@@ -88,7 +88,8 @@
             <EditText
                 android:id="@id/number_text"
                 android:hint="@string/number_hint"
-                android:inputType="number"
+                android:inputType="text"
+                android:digits="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789 "
```

Comment:
Remove `android:digits`. This shouldn't be needed with `inputType=text`.

Reply:
But then it would not be alphanumeric anymore. I thought alphanumeric is only a-z and 0-9?
Otherwise, it will contain @!"$§&/() and so on.

Reply:
Good point. Please write tests to show this works as expected. This should include at least two tests: 

1. Enter a alphanumeric value in the Number field along with valid values in the other fields and assert that the card is created
2. Enter non alphanumeric characters in the Number field and assert the expected error appears.

Reply:
Okay, i would need a little bit of time to figure out how to make tests. And it won't show any error, it would just not allow you to insert any others digits except alpahumeric

Reply:
So, I added a new .csv file containing basically the same test cards, but with the alphanumeric numbers, and edited the already existing test class. There are 2 problems I cannot solve:
1) The check whether new cards are in DB or not doesn't work, even though the toast is appearing.
2) typeTextIntoFocusedView() function doesn't work properly with digits="", it converts "§$%*+# and s.o.  to numbers like 2457